### PR TITLE
[CWS] fix prerm scripts error logs

### PR DESCRIPTION
**Pull request by spikat**
### What does this PR do?

fix prerm scripts to avoid error logs when run/runtime-security/profiles is not empty, otherwise it will generate those error logs:

> rmdir: failed to remove '/opt/datadog-agent/run/runtime-security/profiles': Directory not empty
> rm: cannot remove '/opt/datadog-agent/run/runtime-security/profiles': Is a directory
> rmdir: failed to remove '/opt/datadog-agent/run/runtime-security': Directory not empty
> rm: cannot remove '/opt/datadog-agent/run/runtime-security': Is a directory
> rmdir: failed to remove '/opt/datadog-agent/run': Directory not empty

Errors introduced in this [PR](https://github.com/DataDog/datadog-agent/pull/16763/files)

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]

**Review by KSerrania — COMMENTED** [review]
Kitchen tests are failing after this change:
```
         1) dd-agent-installation-script behaves like an Agent that is removed should remove the installation directory
            Failure/Error: DEFAULT_FAILURE_NOTIFIER = lambda { |failure, _opts| raise failure }
              expected `["/opt/datadog-agent", "/opt/datadog-agent/run", "/opt/datadog-agent/run/sysprobe.sock"].empty?` to be truthy, got false
            Shared Example Group: "an Agent that is removed" called from ./spec_helper.rb:448
            Shared Example Group: "Agent uninstall" called from ./install-script_spec.rb:71
            # ./spec_helper.rb:875:in `block (2 levels) in <top (required)>'
            # /tmp/verifier/gems/gems/busser-rspec_datadog-0.8.5/lib/busser/rspec_datadog/runner.rb:23:in `<main>'
```

Sounds like the `sysprobe.sock` file isn't removed anymore with this new logic.

File: omnibus/package-scripts/agent-deb/prerm [hunk 1]
```diff
@@ -115,16 +115,14 @@ remove_sysprobe_files()
     for file in run/runtime-security-registry.json run/sysprobe.sock run/event-monitor.sock run/runtime-security.sock run/system-probe.pid  run/runtime-security/profiles run/runtime-security
     do
         if [ -d "$INSTALL_DIR/$file" ]; then
-            rmdir "$INSTALL_DIR/$file" || true
-        fi
-
-        if [ -e "$INSTALL_DIR/$file" ]; then
+            rmdir "$INSTALL_DIR/$file" 2>/dev/null || true
```
[the remark this answers is not in the document]
Comment (spikat) [reply omnibus/package-scripts/agent-deb/prerm:118]: Exactly. Sorry if it wasn't clear

[the remark this answers is not in the document]
Comment (spikat) [reply omnibus/package-scripts/agent-deb/prerm:118]: To be more precise: we may have files in run/runtime-security/profiles/ (for security profiles that we want to keep from a version to another one).
I will have a closer look on tests to understand why this changed introduced a regression regarding the sock file.

[the remark this answers is not in the document]
Comment (spikat) [reply omnibus/package-scripts/agent-deb/prerm:127]: ... After team discussion, we finally decided to delete all created files at runtime. Sorry for changing our mind on the way

**Review by safchain — APPROVED** [review]

**Review by bkabrda — COMMENTED** [review]
LGTM, but will require fixing the shellcheck errors. I can approve after these are fixed.

**Comment by spikat** [comment]
> LGTM, but will require fixing the shellcheck errors. I can approve after these are fixed.

Thanks! It should be good now :)

**Review by bkabrda — APPROVED** [review]

**Pull request merged**